### PR TITLE
Automate install process

### PR DESCRIPTION
**Pull request by trotter**
I've added `setup/install` to make it easier to get vcap installed on a VM. It's not perfect at all, but it's much better than going through the manual process.

### CODE REVIEW

**Comment by auser** [comment]
Looks like a good start to me

File: setup/install [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+echo "Installing dependencies"
+if [[ $PLATFORM == 'Linux' ]]; then
+  sudo apt-get -y install coreutils autoconf curl git-core ruby bison build-essential zlib1g-dev libssl-dev libreadline5-dev
+else
+  echo "Sorry, we can't install dependencies for your system yet."
+fi
+
+# Generate random mysql password
+if [[ $PLATFORM == 'Linux' ]]; then
+  MYSQL_PASS="$(date | md5sum | cut -f1 -d' ')"
```
Comment (bjorne) [inline setup/install]: Looks great all in all! However, this password will not be random. How about `cat /dev/urandom | tr -dc _A-Z-a-z-0-9 | head -c10` or something more secure?

Comment (trotter) [inline setup/install]: Oh, I dig that. I'll make the change within a few hours.

**Comment by pbozeman** [comment]
Thanks for the pull request.  We are going to start processing these on Tue and Thur, but this week, we are going to do a pass over these on the 15th.  If we have any questions or comments, we'll let you know then. 

Thanks

p.s.  We are particularly happy to see this one :)

**Comment by auser** [comment]
I am working on a chef cookbook and should have it working tomorrow. Excited to get it actually working and see the internals. More soon. 

Good work trotter, by the way

**Comment by trotter** [comment]
Very nice, is that on github already? I'm more than happy to help with
any chef effort.

On Apr 14, 2011, at 10:38 PM, auser
reply@reply.github.com
wrote:

**Comment by auser** [comment]
Not yet. I'll push it to github tomorrow morning when I get back to my computer. It's almost working, just a few minor updates and it should be golden. I'll message when it's up

**Comment by paulj** [comment]
Once there is a Chef cookbook, it could be wrapped up in a Vagrant configuration too, making it really easy to get an isolated development setup running.

**Comment by auser** [comment]
I added a basic chef cookbook that is not fully working quite yet, but it's up and available on github.

```
https://github.com/auser/cloudfoundry-cookbook
```

**Comment by pbozeman** [comment]
This pull request has some other stuff in it that we don't want to take as is.. (e.g. df0b2fa).  Can you create a feature branch on your repo that starts from the current head of cloundfoundry/vcap and cherry pick over the commits you want from this work and then submit the pull request against that branch.  See:  http://help.github.com/pull-requests/.

Thanks!

**Comment by trotter** [comment]
Whoops, must have started from the wrong branch. Will fix and resubmit soon.

**Comment by trotter** [comment]
Resubmitted as issue #21

**Comment by auser** [comment]
I added a quickstart as we talked about. It's _almost_ completely there. There is an issue when running `rake bundler:install` with cloud_controller, which I have yet to resolve, so perhaps someone could check it out while I'm at work.

Anyway, the quickstart is here:

```
https://github.com/auser/cloudfoundry-quickstart
```

It requires vagrant and VirtualBox and _almost_ completely sets up a cloudfoundry machine.

**Comment by paulj** [comment]
Does the bundler:install need to have rvm use called first? Or does rvm trigger because of the cwd?

**Comment by auser** [comment]
No, rvm is called as a recipe. It's installed before all bundler... the `rake bundler:install` is called to setup vcap

**Comment by paulj** [comment]
Ok. I could see the `rvm use` in front of the `bin/vcap`, so I was expecting something equivalent in front of the `rake bundler:install`, and was wondering if perhaps the gems were going astray. I guess I must be missing something clever going on in the recipe or the Rakefile to setup the RVM environments.

**Comment by auser** [comment]
No, it should be pretty straightforward paulj. Perhaps I'm missing what you mean... there's `jruby -S rake bundler:install` at the end of the default recipe...

It was `execute`, but I changed it to script on the last push or so

https://github.com/auser/cloudfoundry-quickstart/blob/master/chef-repo/cookbooks/cloudfoundry/recipes/default.rb#L175

**Pull request closed without merging**